52.md

```diff
@@ -0,0 +1,142 @@
+NIP-52
+======
+
+Calendar Events
+---------------
+
+`draft` `optional` `author:tyiu`
+
+This specification defines calendar events representing an occurrence at a specific moment or between moments. These calendar events are replaceable and referenceable per [NIP-33](33.md) and deletable per [NIP-09](09.md).
+
+Unlike the term `calendar event` specific to this NIP, the term `event` is used broadly in all the NIPs to describe any Nostr event. The distinction is being made here to discern between the two terms.
+
+## Calendar Events
+
+### Format
+
+The format uses a parameterized replaceable event kind `31923` with a list of tags as described below:
+* `d` (required) universally unique identifier (UUID). Generated by the client creating the calendar event.
+* `summary` (required) name of the calendar event
+* `description` (optional) detailed description of the calendar event
+* `start` (required) inclusive start Unix timestamp in seconds. Must be less than `end`, if it exists.
+* `end` (optional) exclusive end Unix timestamp in seconds
```

Comment:
Kinda think that `end` should be required. Maybe clients should default it to be equivalent to `start` if there is no other value. 
Also, `start` should probably say "Must be less than or equal to `end`."
When querying relays, it would be convenient if relays only need to index one of the fields.

Reply:
Having `end` be optional and exclusive is intentional and inspired by the iCal spec which does the same thing.
https://en.wikipedia.org/wiki/ICalendar#Date_and_time_format

`end` is exclusive so that it doesn't cause scheduling conflicts if another calendar event has `start` at the same time, and also makes it less ambiguous where the end boundary actually is (is it at the beginning of the minute, or is it at the end of the minute). That also means that `start` must be less than `end` if it exists. If `end` is omitted, that means the calendar event is just an instantaneous moment.

Reply:
> Maybe this is insane, but what about making both optional? I.e. for draft calendar events that are not yet scheduled. Not sure what other use cases.

I feel like if something like that is desired, it should be its own kind. As far as traditional calendar software goes, I'm not sure if I've seen that sort of use case before.